test/cli/test-mutate/karpenter-annotations-to-nodeselector/kyverno-test.yaml

```diff
@@ -0,0 +1,17 @@
+name: karpenter-annotations-to-nodeselector
+policies:
+  - policy.yaml
+resources:
+  - resource.yaml
+results:
+  - policy: karpenter-annotations-to-nodeselector
+    rule: hard-nodeselector-lifecycle-on-demand
```

Comment:
```suggestion
    rule: nodeselector-lifecycle-on-demand
```

Reply:
Why this suggestion mvaalexp ?

Reply:
If you are trying to recreate the original issue, this is what causes it:

first result needs to call
https://github.com/kyverno/kyverno/pull/8363/files#diff-6ec7fb9844adcd0991d80bbb31e76e3e1575f3d615fbc31ca96fddee36a456f2R8

which patches
https://github.com/kyverno/kyverno/pull/8363/files#diff-6ec7fb9844adcd0991d80bbb31e76e3e1575f3d615fbc31ca96fddee36a456f2R40

which then triggers the the second rule
https://github.com/kyverno/kyverno/pull/8363/files#diff-6ec7fb9844adcd0991d80bbb31e76e3e1575f3d615fbc31ca96fddee36a456f2R53

Current PR is using only the `hard-nodeselector-lifecycle-on-demand` rule set for both cases (based on your conversation it doesnt matter, but it doesn't match the original test case).

Reply:
Yup